edx_proctoring/api.py

```diff
@@ -494,6 +496,147 @@ def remove_allowance_for_user(exam_id, user_id, key):
         emit_event(exam, 'allowance.deleted', override_data=data)
 
 
+def add_bulk_allowances(exam_ids, user_ids, allowance_type, value):
+    """
+    Adds (or updates) an allowance for multiple users and exams
+    """
+    exam_ids = set(exam_ids)
+    user_ids = set(user_ids)
+
+    log_message = (
+        'Adding allowances of type allowance_type={allowance_type} with value={value} '
+        'for exams exam_ids={exam_ids} and users user_ids={user_ids}'.format(
+            allowance_type=allowance_type,
+            value=value,
+            exam_ids=exam_ids,
+            user_ids=user_ids
+        )
+    )
+    log.info(log_message)
+
+    multiplier = 0
+    if allowance_type == constants.TIME_MULTIPLIER:
+        err_msg = (
+            u'allowance_value "{value}" should be a float value greater than 1.'
+        ).format(value=value)
+        try:
+            multiplier = float(value) - 1
+        except ValueError as error:
+            raise AllowanceValueNotAllowedException(err_msg) from error
+        if multiplier < 0:
+            raise AllowanceValueNotAllowedException(err_msg)
+
+    if allowance_type == constants.ADDITIONAL_TIME:
+        err_msg = (
+            u'allowance_value "{value}" should be a non-negative integer value'
+        ).format(value=value)
+        if not value.isdigit():
+            raise AllowanceValueNotAllowedException(err_msg)
+
+    successes = 0
+    failures = 0
+    data = []
+    if allowance_type == constants.TIME_MULTIPLIER:
+        for exam_id in exam_ids:
+            try:
+                target_exam = get_exam_by_id(exam_id)
+            except ProctoredExamNotFoundException:
+                log_message = (
+                    'Attempted to get exam_id={exam_id}, but this exam does not exist.'.format(exam_id=exam_id)
+                )
+                log.error(log_message)
+                for user_id in user_ids:
+                    failures += 1
+                    data.append({
+
+                            'exam_id': exam_id,
+                            'user_id': user_id,
+                            'status': 400
```

Comment:
Use python HTTP status constants. https://docs.python.org/3/library/http.html#http.HTTPStatus

Reply:
It doesn't looks like this data is used on the view. Do you imaging we should use this on the frontend?
I rather we don't use HTTP status code for frontend. To Simplify, you can omit this, and only when there is clear need on the frontend, we can come back to here and provide the needed data when you are developing the frontend. Sounds good?

Reply:
Yeah that sounds fair, However I was thinking it might be helpful to know which users were added successfully and which weren't, to notify after submitting the allowances in the case that there might have been a typo on one of the users and the instructor has to re add the allowance.